modules/history-utils/history-utils.nu

```diff
@@ -0,0 +1,88 @@
+export-env {
+    $env.history_backup_dir = $'($env.HOME)/.cache/nu-history-backup'
+}
+
+# backup history
+export def 'history backup' [tag?] {
+    ^mkdir -p $env.history_backup_dir
+    let tag = if ($tag | is-empty) { '' } else { $"($tag)::" }
+    [
+        $".output ($env.history_backup_dir)/($tag)(date now | format date "%y_%m_%d_%H_%M_%S").sql"
+        $"update history set cwd = replace\(cwd, '($env.HOME)', '~');"
+        '.dump'
+        $"update history set cwd = replace\(cwd, '~', '($env.HOME)');"
+        '.quit'
+    ]
+    | str join (char newline)
+    | sqlite3 $nu.history-path
+}
+
+def "nu-complete history_backup_file" [] {
+    ls $env.history_backup_dir | each {|x| $x.name | path parse } | get stem | reverse
+}
+# restore history
+export def 'history restore' [name: string@"nu-complete history_backup_file"] {
+    rm -f $nu.history-path
+    cat ([$env.history_backup_dir, $"($name).sql"] | path join)
+    | sqlite3 $nu.history-path
```

Comment:
it would be good to check for sqlite3 before using it. maybe with the which command.

Reply:
There's nothing special about this command. Updates can be done with `query db`.

Reply:
can this sqlite3 be replaced with query db?

Reply:
I'm very sorry, I saw that the reply had a `Pending` tag, and I found that I need to submit it.

Reply:
I've been wondering if the `stor` family of commands can be used to replace the sqlite3 stuff here. For backup and restore, it seems like all you'd have to do is `stor import --file-name $nu.history-path` and then `stor export --file-name /some/other/path/history.sqlite3`. Have you tried that yet?

Reply:
It seems feasible, but the database file is larger than the sql file, but it is acceptable.
The main question is, if other applications are also using stor and create tables, will `stor import` delete the existing tables at this time, and if there are redundant tables when saving, will they be saved together?

Reply:
If it is not limited to text format, can it just copy the database file?
Text formats can be parsed and processed, and there may be some scenarios where this is required.

Reply:
There seems to be a small problem with stor
```
stor create -t foo -c { bar: str }
stor insert -t foo -d {bar: "foobar"}
stor insert -t foo -d {bar: "foo'bar"}
Error:   × Failed to open SQLite connection in memory from insert
   ╭─[source:1:1]
 1 │ nu
   · ▲
   · ╰── near "bar": syntax error in INSERT INTO foo ( bar) VALUES ( 'foo'bar') at offset 37
   ╰────

stor insert -t foo -d {bar: "foo\'bar"}
Error:   × Failed to open SQLite connection in memory from insert
   ╭─[source:1:1]
 1 │ nu
   · ▲
   · ╰── near "bar": syntax error in INSERT INTO foo ( bar) VALUES ( 'foo'bar') at offset 37
   ╰────

stor insert -t foo -d {bar: "foo\\'bar"}
Error:   × Failed to open SQLite connection in memory from insert
   ╭─[source:1:1]
 1 │ nu
   · ▲
   · ╰── near "bar": syntax error in INSERT INTO foo ( bar) VALUES ( 'foo\'bar') at offset 38
   ╰────

```